assets/app/scripts/services/data.js

```diff
@@ -360,6 +360,51 @@ angular.module('openshiftConsole')
     return deferred.promise;
   };
 
+// https://developer.mozilla.org/en-US/docs/Web/API/WindowBase64/btoa
+function utf8_to_b64( str ) {
+    return window.btoa(window.unescape(encodeURIComponent( str )));
+}
+function b64_to_utf8( str ) {
+    return decodeURIComponent(window.escape(window.atob( str )));
+}
+
+// TODO (bpeterse): ideally DataService would be more general & log specific function
+// could be in a separate service, but we need the helper functions
+DataService.prototype.watchLogStream = function(kind, name, context, onMessage, onClose) {
```

Comment:
an option to this should be whether we want the base64 encode/decode behavior... if so, it sets that protocol and attempts the auto-decode... otherwise, the caller just gets the raw websocket input/output